src/RabbitMQ/AmqpTemplate.php

```diff
@@ -18,16 +22,63 @@
     protected $channel;
 
     /**
+     * @var string
+     */
+    protected $escapeMode;
+
+    /**
      * Constructor
      *
      * @param string $host
      * @param int    $port
      * @param string $user
      * @param string $pass
+     * @param string $escapeMode
      */
-    public function __construct($host, $port, $user, $pass)
+    public function __construct($host, $port, $user, $pass, $escapeMode = self::ESCAPE_MODE_SERIALIZE)
     {
         $this->connection = new AMQPConnection($host, $port, $user, $pass);
         $this->channel = $this->connection->channel();
+        $this->escapeMode = $escapeMode;
+    }
+
+    /**
+     * Escape the message
+     *
+     * @param $message
+     * @return string
+     */
+    protected function escape($message)
```

Comment:
hmmm... why don't you use a factory to provide the escaper depending on the escape mode instead of an hardcoded switch ? More config, but less coupling with impl details.
Unless you think there won't be any other serializers because those two solve 99% cases (sounds true) :)